linetools/spectra/io.py

```diff
@@ -473,60 +495,60 @@ def parse_linetools_spectrum_format(hdulist):
     xspec1d : XSpectrum1D
       Parsed spectrum
 
-
     """
     if 'WAVELENGTH' not in hdulist:
-        spec1d = spec_read_fits.read_fits_spectrum1d(
-            os.path.expanduser(datfil), dispersion_unit='AA')
+        pdb.set_trace()
+        #spec1d = spec_read_fits.read_fits_spectrum1d(
+        #    os.path.expanduser(datfil), dispersion_unit='AA')
         xspec1d = XSpectrum1D.from_spec1d(spec1d)
     else:
         wave = hdulist['WAVELENGTH'].data * u.AA
         fx = hdulist['FLUX'].data
-        xspec1d = XSpectrum1D.from_tuple((wave, fx, None, None))
 
     # Error array
     if 'ERROR' in hdulist:
         sig = hdulist['ERROR'].data
-        xspec1d.uncertainty = StdDevUncertainty(sig)
     else:
-        xspec1d.uncertainty = None
+        sig = None
 
     if 'CONTINUUM' in hdulist:
-        xspec1d.co = hdulist['CONTINUUM'].data
+        co = hdulist['CONTINUUM'].data
     else:
-        xspec1d.co = None
+        co = None
+
+    xspec1d = XSpectrum1D.from_tuple((wave, fx, sig, co))
 
     if 'METADATA' in hdulist[0].header:
         xspec1d.meta.update(json.loads(hdulist[0].header['METADATA']))
 
     return xspec1d
 
-def parse_DESI_brick(hdulist, exten=None):
+def parse_DESI_brick(hdulist, select=0):
     """ Read a spectrum from a DESI brick format HDU list
 
     Parameters
     ----------
     hdulist : FITS HDU list
-    exten : int, optional
-      Extension for the date. Default is 0 (the primary HDU).
+    select : int, optional
+      Spectrum selected. Default is 0
 
     Returns
     -------
     xspec1d : XSpectrum1D
       Parsed spectrum
     """
-    if exten is None:
-        exten = 0
-    fx = hdulist[0].data[exten, :]
+    fx = hdulist[0].data
     # Sig
-    ivar = hdulist[1].data[exten, :]
-    sig = np.zeros(ivar.size)
+    ivar = hdulist[1].data
+    sig = np.zeros_like(ivar)
     gdi = ivar > 0.
     sig[gdi] = np.sqrt(1./ivar[gdi])
     # Wave
     wave = hdulist[2].data
     wave = give_wv_units(wave)
-    xspec1d = XSpectrum1D.from_tuple((wave, fx, sig, None))
+    wave = np.outer(np.ones(fx.shape[0]), wave)
```

Comment:
Maybe replace this with `np.tile(wave, (fx.shape[0],1))`  which looks clearer to me (and is a bit faster).